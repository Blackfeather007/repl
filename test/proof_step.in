{"cmd" : "def f : Nat := by sorry"}

{"tactic": "apply Int.natAbs", "proofState": 0}

{"tactic": "have t := 42", "proofState": 0}

{"tactic": "exact t", "proofState": 2}
