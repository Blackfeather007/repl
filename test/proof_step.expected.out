{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 18},
   "goal": "⊢ Nat",
   "endPos": {"line": 1, "column": 23}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 4},
   "endPos": {"line": 1, "column": 5},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofState": 1, "messages": [], "goals": ["⊢ Int"]}

{"proofState": 2, "messages": [], "goals": ["t : Nat\n⊢ Nat"]}

{"proofState": 3, "messages": [], "goals": []}

